### PR TITLE
Adding typings for fs-extra-promise-es6 (sourced from fs-extra-promise) [types-2.0]

### PR DESCRIPTION
**Pull request by geoffreak**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

---

Definitions for [`fs-extra-promise-es6`](https://github.com/vinsonchuong/fs-extra-promise-es6). Typings borrowed and tweaked from `fs-extra-promise`, which has the same API, but uses Bluebird instead of native promises.

### CODE REVIEW

**Comment by andy-ms** [comment]
Could this reuse `Stats` from the `node` typings?

**Comment by geoffreak** [comment]
@andy-ms It certainly seems that way. I have also noticed the typings were out of date for the read/write streams and have updated those to match the `fs` typing. It's too bad you can't inherit the typing from a module.

File: fs-extra-promise-es6/index.d.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@
 import stream = require("stream");
 import fs = require("fs");
 
+export interface Stats extends fs.Stats {
```
Comment (andy-ms) [inline fs-extra-promise-es6/index.d.ts]: You could use a type alias for this: `export type Stats = fs.Stats`.

Comment (geoffreak) [inline fs-extra-promise-es6/index.d.ts]: Okay, I'll switch to that

**Comment by andy-ms** [comment]
Some of the failing tests aren't due to this PR - merge master into this branch to get the fix.

**Comment by geoffreak** [comment]
I have rebased this branch on the latest `types-2.0` branch. Do you think that would be sufficient? I'd really rather not have to deal with 135 merge conflicts that `master` is reporting in other typings.

**Comment by andy-ms** [comment]
Sorry, meant `types-2.0`.

**Comment by geoffreak** [comment]
@andy-ms Forgive my ignorance, but how often are the npm repos updated? I'm trying to determine when I should expect to see the accepted PRs in npm.

**Comment by andy-ms** [comment]
@geoffreak should be sometime today.

**Pull request merged**